crates/isograph_schema/src/validate_schema.rs

```diff
@@ -711,11 +758,16 @@ fn validate_client_field_definition_selection_exists_and_type_matches(
                         linked_field_selection,
                         server_fields,
                         client_field_args,
+                        &mut used_variables2,
                     )
                 },
             )
         })
-    })
+    });
+
+    used_variables.append(&mut used_variables2);
```

Comment:
This annoys me, but I've run into this issue before as well. This is what I would expect us to do.